### PR TITLE
[onert] Adds an acl_neon kernel for OneHot operation

### PR DESCRIPTION
**Pull request by ragmani**
For issue #4223

This commit adds an acl_neon kernel for OneHot operation.
  - Introduces NEOneHot and NEOneHotKernel into ArmComputeEx
  - Applies the acl_neon kernel

Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

**Comment by ragmani** [comment]
This PR also improves the performance of OneHot operation on acl_neon backend

- With as-is kernel
MEAN     :  1.828 ms
```bash
$ BACKENDS="acl_neon" ./Product/armv7l-linux.release/out/bin/nnpackage_run --nnpackage onehot_4x10137 -w 10 -r 1000
Package Filename onehot_4x10137
===================================
MODEL_LOAD   takes 1.957 ms
PREPARE      takes 83.812 ms
EXECUTE      takes 1.828 ms
- MEAN     :  1.828 ms
- MAX      :  1.946 ms
- MIN      :  1.495 ms
- GEOMEAN  :  1.826 ms
===================================
```

- With this kernel
MEAN     :  0.111 ms
```bash
$ BACKENDS="acl_neon" ./Product/armv7l-linux.release/out/bin/nnpackage_run --nnpackage onehot_4x10137 -w 10 -r 1000
Package Filename onehot_4x10137
===================================
MODEL_LOAD   takes 1.849 ms
PREPARE      takes 82.892 ms
EXECUTE      takes 0.111 ms
- MEAN     :  0.111 ms
- MAX      :  0.295 ms
- MIN      :  0.088 ms
- GEOMEAN  :  0.110 ms
===================================
```

**Review by hseok-oh — APPROVED** [review]
LGTM

**Comment by ragmani** [comment]
@glistening PTAL

**Review by Hyunjun85 — APPROVED** [review]

**Comment by ragmani** [comment]
@glistening PTAL

File: compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+      [&](const Coordinates &id) {
+        std::fill_n(output_it.ptr(),
+                    _output->info()->dimension(0) * _output->info()->element_size(), off_value);
+        Coordinates indices_id(id);
+        indices_id.remove(0);
+        const auto new_index = *(reinterpret_cast<U *>(_indices->ptr_to_element(indices_id)));
+        Coordinates onehot_id(id);
+        onehot_id.set(0, new_index);
+        std::copy_n(_on_value->buffer(), _output->info()->element_size(),
+                    _output->ptr_to_element(onehot_id));
```
Comment (glistening) [inline compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp:122]: @ragmani What happens when `index = -1`?

Comment (ragmani) [reply compute/ARMComputeEx/src/core/NEON/kernels/NEOneHotKernel.cpp:122]: Oh, I missed it. I will fix it. Thanks for your information.

**Review by glistening — APPROVED** [review]
LGTM. Please update index = -1 case in next PR.

**Pull request merged**